Applications/Spire/Source/Charting/DropDownColorPicker.cpp

```diff
@@ -0,0 +1,110 @@
+#include "Spire/Charting/DropDownColorPicker.hpp"
+#include <QEvent>
+#include <QHBoxLayout>
+#include <QMouseEvent>
+#include <QPainter>
+#include "Spire/Charting/ColorPicker.hpp"
+#include "Spire/Spire/Dimensions.hpp"
+
+using namespace boost::signals2;
+using namespace Spire;
+
+DropDownColorPicker::DropDownColorPicker(QWidget* parent)
+    : QWidget(parent),
+      m_has_mouse(false) {
+  setMouseTracking(true);
+  setFocusPolicy(Qt::StrongFocus);
+  window()->installEventFilter(this);
```

Comment:
Is there an alternative to installing an event filter.

Reply:
It does't appear so. There's no signal to just listen for a resize or move.

Reply:
Okay leave as is.